advanced/spec/binarySearchTree_test.js

```diff
@@ -0,0 +1,31 @@
+import chai, { expect } from 'chai'
+import chaiChange from 'chai-change'
+import BinarySearchTree from '../src/binarySearchTree'
+
+chai.use(chaiChange)
+
+describe('BinarySearchTree', () => {
+'use strict'
+
+  it('exists', () => {
+    expect(BinarySearchTree).to.be.a('function')
+  })
+
+  it('should insert node with specified value into tree', () => {
```

Comment:
Instead of starting with an it statement, there should be a describe that tells the person running the test which function is being tested. Something like: 
```
describe(  '#insert', function() {
   context( 'when given the parameter \'Lights\', ', function() {
      it( 'should insert node with specified value into tree', function() {
      })
   })
})
```
You can provide a context to be more descriptive, or you can just choose to use the 'it' statement. Also, mocha strongly encourages using ```function() { }``` instead of ```() => { }```. Here is an explanation here: https://github.com/mochajs/mocha/issues/2018
Jared brought this to my attention many weeks ago, so I follow this rule to avoid potential issues.